framework/src/base/ExecFlagRegistry.C

```diff
@@ -8,31 +8,19 @@
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
 #include "ExecFlagRegistry.h"
-
-#include <mutex>
-
 #include "MooseUtils.h"
 
-namespace moose
-{
-namespace internal
-{
-
 ExecFlagRegistry &
-getExecFlagRegistry()
+ExecFlagRegistry::getExecFlagRegistry()
 {
   static ExecFlagRegistry exec_flag_registry;
   return exec_flag_registry;
 }
 
-ExecFlagRegistry::ExecFlagRegistry() {}
-
 const ExecFlagType &
 ExecFlagRegistry::registerFlag(const std::string & name, const bool is_default)
 {
   const auto name_upper = MooseUtils::toUpper(name);
-  std::unique_lock lock(_flags_mutex);
```

Comment:
Do we no longer have threading concerns?

Reply:
From the Scott Meyer's pattern, the use of a "function static" for the singleton pattern removes that concern. Statics are allocated in the "base" storage (below the heap). There's no possible race condition with this pattern.